### ISSUE TITLE
[CLI] Offline bundle on iOS9-beta2 (Xcode-7-beta)

### ISSUE THREAD

**Issue by ecesena** [facebook/react-native#1751]
I'm not able to run react native apps with offline bundle on iOS9.
Both my own app, and the Examples/SampleApp. They both work accessing the development server.

Repro:
  1. Pull react-native and npm install
  1. Got to examples/SampleApp, add a simple package.json with react-native 0.6.0 dep
  1. Open SampleApp in Xcode, run on simulator or device - it works
  1. Run react-native bundle --minify
  1. Run the app with offline bundle in the simulator - it fails with error "Unexpected identifier 'transformMatrix'".

By changing the way I bundle and wether I try to run on simulator or device I get different the errors.

**Comment by ide** [comment]
Can you try 0.7.0-rc or master? May be fixed by https://github.com/facebook/react-native/pull/1672

**Comment by ecesena** [comment]
Not sure I did the right thing - I now have this package.json (without package.json, I can't bundle).
```
{
  "name": "react-native-sample",
  "dependencies": {
    "react-native": "git+https://github.com/facebook/react-native.git"
  }
}
```
  1. Deleted node_modules/react-native, then run npm install.
  1. Open Xcode and run the app with dev server - it works
  1. bundle fails to compile
```
/Users/ec/devel/react-native/Examples/SampleApp/node_modules/react-native/node_modules/babel-core/lib/babel/tools/protect.js:15
    throw new Error("Don't hotlink internal Babel files.");
...
```

Any better way to set the dep to 0.7.0-rc or master?

**Comment by ide** [comment]
mmm.. not sure. I think there are some issues with Babel in the release (namely the babel version doesn't match the babel-core version) that might need to be addressed.

**Comment by ecesena** [comment]
All right, I'll wait a few more commits then retry...

**Comment by frantic** [comment]
Just tried this:

```
react-native init BabelSample
cd BabelSample
rm -r node_modules
npm install react-native git+https://github.com/facebook/react-native.git --save
react-native bundle
```

and it successfully created a bundle.

Btw if you want to help, add `react-native bundle` test to end2end tests :)

**Comment by landakram** [comment]
I'm running into this same issue: when I try to bundle with `--minify`, the app crashes on iOS 9. Getting this error referencing `transformMatrix`: 

```
2015-07-15 18:35:25.497 [error][tid:com.facebook.React.JavaScript] 'Error: Unexpected identifier \'transformMatrix\'. Expected \']\' to end a subscript expression.\n stack: \n  A          main.jsbundle:10\n  <unknown>  main.jsbundle:10\n  r          main.jsbundle:1\n  <unknown>  main.jsbundle:9\n  r          main.jsbundle:1\n  <unknown>  main.jsbundle:2\n  r          main.jsbundle:1\n  <unknown>  main.jsbundle:1\n  r          main.jsbundle:1\n  r          main.jsbundle:1\n  <unknown>  main.jsbundle:17\n URL: file:///var/mobile/Containers/Bundle/Application/<redacted>/main.jsbundle\n line: 54\n message: Unexpected identifier \'transformMatrix\'. Expected \']\' to end a subscript expression.'
2015-07-15 18:35:25.543 [info][tid:com.facebook.React.JavaScript] 'Running application "MyAppName" with appParams: {"rootTag":1,"initialProps":{}}. __DEV__ === false, development-level warning are OFF, performance optimizations are ON'
2015-07-15 18:35:25.547 [error][tid:com.facebook.React.JavaScript] 'Error: Invariant Violation: Application MyAppName has not been registered.\n stack: \n  runApplication  main.jsbundle:13\n  f               main.jsbundle:3\n  _callFunction   main.jsbundle:3\n  <unknown>       main.jsbundle:3\n  <unknown>       main.jsbundle:3\n  perform         main.jsbundle:3\n  batchedUpdates  main.jsbundle:5\n  u               main.jsbundle:2\n  <unknown>       main.jsbundle:3\n  R               main.jsbundle:3\n  S               main.jsbundle:3\n  processBatch    main.jsbundle:3\n URL: file:///var/mobile/Containers/Bundle/Application/<redacted>/main.jsbundle\n line: 2\n message: Invariant Violation: Application MyAppName has not been registered.'
```

I upgraded to `react-native: 0.7.1` and then also tried the steps above from @frantic, but I'm still encountering the error (it creates the bundle fine, but then crashes during run time).

**Comment by ecesena** [comment]
Since 0.8.0rc it works for me (0.7.x not tested) - should we close this?

**Comment by sahrens** [comment]
Yeah, we hit this internally and fixed it.

**Comment by Fanghao** [comment]
Got the same issue ""Error: Unexpected identifier 'transformMatrix'. Expected ']' to end a subscript expression."

@sahrens May I know which commit fix it? wanna run 0.6-stable on iOS9.

**Comment by landakram** [comment]
Thanks y'all -- turns out I was upgrading react via `npm install`, but not the CocoaPod... :astonished:
